presto-hive/src/main/java/io/prestosql/plugin/hive/security/SqlStandardAccessControlMetadata.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.hive.security;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import io.prestosql.plugin.hive.metastore.HivePrincipal;
+import io.prestosql.plugin.hive.metastore.HivePrivilegeInfo;
+import io.prestosql.plugin.hive.metastore.SemiTransactionalHiveMetastore;
+import io.prestosql.plugin.hive.metastore.thrift.ThriftMetastoreUtil;
+import io.prestosql.spi.PrestoException;
+import io.prestosql.spi.connector.ConnectorSession;
+import io.prestosql.spi.connector.SchemaTableName;
+import io.prestosql.spi.security.GrantInfo;
+import io.prestosql.spi.security.Privilege;
+import io.prestosql.spi.security.PrivilegeInfo;
+import io.prestosql.spi.security.RoleGrant;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static io.prestosql.plugin.hive.metastore.HivePrivilegeInfo.toHivePrivilege;
+import static io.prestosql.plugin.hive.metastore.thrift.ThriftMetastoreUtil.listEnabledPrincipals;
+import static io.prestosql.plugin.hive.security.SqlStandardAccessControl.ADMIN_ROLE_NAME;
+import static io.prestosql.spi.StandardErrorCode.ALREADY_EXISTS;
+import static io.prestosql.spi.security.PrincipalType.USER;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toSet;
+
+public class SqlStandardAccessControlMetadata
+        implements AccessControlMetadata
+{
+    private static final Set<String> RESERVED_ROLES = ImmutableSet.of("all", "default", "none");
+
+    private final SemiTransactionalHiveMetastore metastore;
+
+    public SqlStandardAccessControlMetadata(SemiTransactionalHiveMetastore metastore)
+    {
+        this.metastore = requireNonNull(metastore, "metastore is null");
+    }
+
+    @Override
+    public void createRole(ConnectorSession session, String role, Optional<HivePrincipal> grantor)
+    {
+        // roles are case insensitive in Hive
```

Comment:
This comment is confusing, because the code here seems to be case sensitive.

Reply:
Correct. It looks like a bug that existed before. Let me fix that in separate PR.

Reply:
https://github.com/prestosql/presto/pull/1077